### ISSUE TITLE
Serializing new tree causes NullPointerException when saveTreeState enabled

### ISSUE THREAD

**Issue by bwzhang96** [aws/random-cut-forest-by-aws#142]
On the compact-rcf branch, if you try to map a newly created forest using a mapper with saveTreeState enabled, it will cause a NullPointerException

Example test:
```
    @Test
    public void npe() {
        int dimensions = 4;
        int numberOfTrees = 50;
        int sampleSize = 256;
        Precision precision = Precision.DOUBLE;

        RandomCutForest forest = RandomCutForest.builder().compactEnabled(true).dimensions(dimensions)
            .numberOfTrees(numberOfTrees).sampleSize(sampleSize).precision(precision).build();

        RandomCutForestMapper mapper = new RandomCutForestMapper();
        mapper.setSaveExecutorContext(true);
        mapper.setCopy(true);
        mapper.setSaveTreeState(true);

        RandomCutForestState state = mapper.toState(forest);
    }
```

Stacktrace:
```
java.lang.NullPointerException
	at com.amazon.randomcutforest.tree.AbstractCompactRandomCutTree.getRootIndex(AbstractCompactRandomCutTree.java:205)
	at com.amazon.randomcutforest.state.tree.CompactRandomCutTreeDoubleMapper.toState(CompactRandomCutTreeDoubleMapper.java:70)
	at com.amazon.randomcutforest.state.RandomCutForestMapper.lambda$toState$1(RandomCutForestMapper.java:169)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1384)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:566)
	at com.amazon.randomcutforest.state.RandomCutForestMapper.toState(RandomCutForestMapper.java:169)
```

**Comment by jotok** [comment]
Thanks. It looks like all our tests used nonempty trees and we didn't account for the case where the tree root would be null.

**Comment by jotok** [comment]
Closed by #146
